src/inspector/models/score/scoreaccessibilitysettingsmodel.cpp

```diff
@@ -0,0 +1,214 @@
+                {
+                    value,
+                    QVariantMap {
+                        { preset, i },
+                        { edited, true }
+                    }
+                }
+            });
+        }
+    }
+
+    return presets;
+}
+
+void ScoreAccessibilitySettingsModel::setScoreStylePresetIndex(int index)
+{
+    bool isEdited = false;
+    ScoreStylePreset selectedPreset;
+
+    if (m_scoreStylePresetEdited) {
+        if (index == static_cast<int>(m_scoreStylePreset) + 1) {
+            isEdited = true;
+        }
+        selectedPreset = static_cast<ScoreStylePreset>(index - (isEdited ? 1 : 0));
+    } else {
+        selectedPreset = static_cast<ScoreStylePreset>(index);
+    }
+
+    bool presetChanged = (selectedPreset != m_scoreStylePreset);
+    bool editedChanged = (isEdited != m_scoreStylePresetEdited);
+
+    if (presetChanged || editedChanged) {
+        m_scoreStylePreset = selectedPreset;
+        m_scoreStylePresetEdited = isEdited;
+
+        if (!isEdited) {
+            loadStyle(static_cast<int>(m_scoreStylePreset));
+        }
+
+        emit possibleScoreStylePresetsChanged();
+        emit scoreStylePresetIndexChanged();
+    }
+}
+
+void ScoreAccessibilitySettingsModel::loadStyle(int preset)
+{
+    ScoreStylePreset presetName = static_cast<ScoreStylePreset>(preset);
+    muse::io::path_t filePath = getStyleFilePath(presetName);
+    m_ignoreStyleChange = true;
+
+    if (presetName == ScoreStylePreset::DEFAULT) {
+        if (filePath.empty()) {
+            StyleIdSet emptySet;
+            globalContext()->currentNotation()->style()->resetAllStyleValues(emptySet);
+        } else {
+            globalContext()->currentNotation()->style()->loadStyle(filePath, true);
+        }
+    } else if (!filePath.empty()) {
+        LOGI() << "Loading style from filePath:" << filePath;
+        globalContext()->currentNotation()->style()->loadStyle(filePath, true);
+    } else {
+        LOGI() << "filePath is empty";
+    }
+    globalContext()->currentNotation()->elements()->msScore()->score()->style().setStylePreset(presetName);
+
+    m_scoreStylePresetEdited = false;
+    globalContext()->currentNotation()->elements()->msScore()->score()->style().setStylePresetEdited(false);
+    emit possibleScoreStylePresetsChanged();
+    emit scoreStylePresetIndexChanged();
+    m_ignoreStyleChange = false;
+}
+
+int ScoreAccessibilitySettingsModel::scoreStylePresetIndex() const
+{
+    return m_scoreStylePresetEdited ? static_cast<int>(m_scoreStylePreset) + 1 : static_cast<int>(m_scoreStylePreset);
+}
+
+void ScoreAccessibilitySettingsModel::updateScoreStylePreset()
+{
+    MStyle style = globalContext()->currentNotation()->elements()->msScore()->score()->style();
+    ScoreStylePreset stylePreset = style.stylePreset();
+    bool stylePresetEdited = style.stylePresetEdited();
+    bool indexChanged = false;
+
+    if (m_scoreStylePreset != stylePreset) {
+        m_scoreStylePreset = stylePreset;
```

Comment:
If you change the cached value (`m_scoreStylePreset `), you also need to update the real value (`style.setStylePreset()`).

Reply:
I should see:

```C++
m_scoreStylePreset = stylePreset;
style.setStylePreset(stylePreset);
```

Reply:
The reason I didn't add this setter is that `stylePreset` is essentially the same as `style.preset()`. Since we are using the style object values to correctly update the dropdown, updating the style object value again with the dropdown value can be skipped. The same is the idea for `stylePresetEdited`

Reply:
Oh yes, of course. How silly of me! You literally just [read that value](https://github.com/musescore/MuseScore/pull/23048/files#diff-68cac91f52e04f14d17e9fa1928b497c8f5aaf9cf08a2ea8fb3006807b404e7eR198), so there's no point setting it again.